src/lib.rs

```diff
@@ -60,17 +208,81 @@ impl Command {
         )
     }
 
+    /// Executes a command as a child process, waiting for it to finish and
+    /// collecting its status.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use CreateProcessW::Command;
+    ///
+    /// let status = Command::new("notepad.exe")
+    ///     .status()
+    ///     .expect("failed to execute process");
+    ///
+    /// println!("process finished with: {}", status.code());
+    ///
+    /// assert!(status.success());
+    /// ```
     pub fn status(&mut self) -> Result<ExitStatus> {
         self.spawn()?.wait()
     }
 }
 
+use std::ffi::{c_void, OsStr};
+use std::mem::size_of;
+use std::path::Path;
+use windows::Win32::Foundation::{GetLastError, PWSTR, STATUS_PENDING};
+use windows::Win32::Security::SECURITY_ATTRIBUTES;
+use windows::Win32::System::Threading::{
+    GetExitCodeProcess, GetProcessId, TerminateProcess, WaitForSingleObject,
+    PROCESS_CREATION_FLAGS, PROCESS_INFORMATION, STARTUPINFOW, WAIT_OBJECT_0,
+};
+use windows::Win32::System::WindowsProgramming::INFINITE;
+
+/// Representation of a running or exited child process.
+///
+/// This structure is used to represent and manage child processes. A child
+/// process is created via the [`Command`] struct, which configures the spawning
+/// process and can itself be constructed using a builder-style interface.
+///
+/// # Warnings
+///
+/// Calling [`wait`][Child::wait] is necessary for the OS to release resources.
+/// A process that terminated but has not been waited on is still around as a
+/// "zombie". Leaving too many zombies around may exhaust global resources.
+///
+/// This library does *not* automatically wait on child processes (not even if
+/// the `Child` is dropped), it is up to the application developer to do so. As
+/// a consequence, dropping `Child` handles without waiting on them first is not
+/// recommended in long-running applications.
+///
+/// # Examples
+///
+/// ```no_run
+/// use CreateProcessW::Command;
+///
+/// let mut child = Command::new("notepad.exe")
+///     .spawn()
+///     .expect("failed to execute child");
+///
+/// let status = child.wait().expect("failed to wait on child");
+///
+/// assert!(status.success());
+/// ```
 #[derive(Debug)]
 pub struct Child {
     process_information: PROCESS_INFORMATION,
 }
 
 impl Child {
+    #[doc(hidden)]
+    /// Create a new process and initialize it's memory. If it cannot be
+    /// created, an [`CreateFailed`][Error::CreateFailed] error is returned.
+    ///
+    /// Equivalent to [`CreateProcessW`][create-process-w]
+    ///
+    /// [create-process-w]: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw
```

Comment:
1. Doc comments on private functions are always hidden.
2. Turn doc comments to comments.

```suggestion
    // Create a new process and initialize it's memory. If it cannot be
    // created, an [`CreateFailed`][Error::CreateFailed] error is returned.
    //
    // Equivalent to [`CreateProcessW`][create-process-w]
    //
    // [create-process-w]: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw
```